rockcraft/models/project.py

```diff
@@ -342,9 +343,23 @@ def _validate_entrypoint_service(
 
         if entrypoint_service not in values.get("services", {}):
             raise ProjectValidationError(
-                f"The provided entrypoint-service '{entrypoint_service}' is not a valid Pebble service."
+                f"The provided entrypoint-service '{entrypoint_service}' is not "
+                + "a valid Pebble service."
             )
 
+        command = values.get("services")[entrypoint_service].command
+        command_sh_args = shlex.split(command)
+        # optional arg is surrounded by brackets, so check that they exist in the
+        # right order
+        try:
+            if command_sh_args.index("[") >= command_sh_args.index("]"):
+                raise ValueError
```

Comment:
ProjectValidationError will apparently need an argument, so the error string will be duplicated here too.
Check: https://github.com/canonical/rockcraft/actions/runs/7004567791/job/19052578345

Reply:
Noted. We need a different strategy then, cause `ValueError` is not the right exception in this case. Let's break it down:
 - if either `[` or `]` don't exit, this code won't get to `raise ValueError`, and instead raise that exception by itself;
 - if it gets to line 356 (`raise ValueError`), then it's because the indexed of either `[` or `]` are wrong
 - in both cases, we want to raise a ProjectValidationError

I'd say: either raise `IndexError` or `ProjectValidationError` with an explanatory message.